src/helpers.rs

```diff
@@ -616,6 +616,12 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         Ok(wchars)
     }
+
+    fn in_std(&self) -> bool {
+        let this = self.eval_context_ref();
+        this.tcx.def_path(this.frame().instance.def_id()).krate
+            == this.tcx.def_path(this.tcx.lang_items().start_fn().unwrap()).krate
```

Comment:
There may not be a `#[lang = "start"]` when using `#![no_std]` in combination with `#[start]`, so please don't unwrap.

Reply:
There is already an `unwrap()` in `create_ecx()` :thinking:
https://github.com/rust-lang/miri/blob/ef998308805748289e46bce394a5d539b12b0118/src/eval.rs#L113

Reply:
> using #![no_std] in combination with #[start]

That sounds like a test case we should have, then. ;)

Reply:
> using `#![no_std]` in combination with `#[start]`

Note that it's currently broken even before the `unwrap()`:
```
error: internal compiler error: src/tools/miri/src/eval.rs:110:9: main function must not take any arguments

thread 'rustc' panicked at 'Box<Any>', compiler/rustc_errors/src/lib.rs:1007:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.54.0-nightly (c79419af0 2021-06-04) running on x86_64-unknown-linux-gnu
```
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=b867b78dd787c573bdfb44c88fa856dc
https://github.com/rust-lang/miri/blob/ef998308805748289e46bce394a5d539b12b0118/src/eval.rs#L110

Reply:
Hm, the start item in Rust has no arguments so I am not sure if it makes sense to support other start items that do.

But this sounds like a separate discussion then, could you open an issue?